tests/integration/test_npm.py

```diff
@@ -30,6 +30,15 @@
             ),
             id="npm_lockfile3_yarn_registry",
         ),
+        pytest.param(
+            utils.TestParameters(
+                repo="https://github.com/cachito-testing/npm-dev-optional-peer-deps.git",
+                ref="346daddf5bda914485ead4f957901912796916d1",
+                packages=({"path": ".", "type": "npm"},),
+                check_vendor_checksums=False,
+            ),
+            id="npm_dev_optional_peer_deps",
```

Comment:
I dropped the `lockfile3` identification we're using in other integration tests, since we will only use this lockfile version for this test. This makes it somewhat out of the standard though. Maybe we should rename the other tests when we fully drop v1 support?

I also only made a single branch in the test repo, main. Let me know if you're ok with this.